### PR TITLE
Adding pdal-c package 

### PR DESCRIPTION
**Pull request by runette**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by runette** [comment]
@conda-forge-admin, please ping team

This is a new package containing the C API for PDAL - which already a Conda-Forge package.

It is probably most suitable for the c/c++ team to review.

Thanks

Paul

File: recipes/pdal-c/build.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  -DCMAKE_INSTALL_PREFIX=$PREFIX \
+  -DCMAKE_LIBRARY_PATH=$PREFIX/lib \
+  -DCMAKE_INCLUDE_PATH=$PREFIX/include \
+  -DCONDA_BUILD=ON \
+  .
+  
+
+# CircleCI offers two cores.
+make -j $CPU_COUNT
+make install
```
Comment (ocefpaf) [inline recipes/pdal-c/build.sh:21]: Do we have a make check we can run here?

Comment (runette) [reply recipes/pdal-c/build.sh:21]: The short answer is no ... We don't currently have any and this recipe was adapted from the parent PDAL recipe.

What are you expecting to see?

Thanks for picking up the review.

Paul

Comment (ocefpaf) [reply recipes/pdal-c/build.sh:21]: Just more tests if they are available. It is fine if we don't have them but if we do, let's run them.

**Pull request merged**